packages/react/src/DevMode.tsx

```diff
@@ -0,0 +1,110 @@
+import { type ComponentProps, type ComponentType, useState } from 'react'
+import { useDevModeObserve } from './hooks'
+import { noop } from './utils'
+
+export type PropsWithoutDevMode<TProps extends ComponentProps<ComponentType>> = Omit<TProps, 'devMode'>
+
+export type PropsWithDevMode<
+  TDevModeProps extends Record<string, any>,
+  TComponentProps extends ComponentProps<ComponentType>,
+> = TComponentProps & {
+  /**
+   * @experimental This is experimental feature.
+   */
+  devMode?: TDevModeProps
+}
+
+type Sync = () => void
+class SuspensiveDevMode {
+  constructor(public is = false) {}
```

Comment:
SuspensiveDevMode.is will be always false before developer's clicking `<DevMode/>`.
This will prevent misunderstanding if SuspensiveDevMode on or off now.